### PR TITLE
chore: added InterchainTokenIdClaimed event

### PR DESCRIPTION
**Pull request by ahramy**
[AXE-5210](https://axelarnetwork.atlassian.net/browse/AXE-5210)
- added `InterchainTokenIdClaimed` event
- `coverage` remains the same. 

[AXE-5210]: https://axelarnetwork.atlassian.net/browse/AXE-5210?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: contracts/InterchainTokenService.sol [hunk 1]
```diff
@@ -341,6 +341,8 @@ contract InterchainTokenService is
 
         tokenId = interchainTokenId(deployer, salt);
 
+        emit InterchainTokenIdClaimed(tokenId, deployer, salt);
```
Comment (milapsheth) [inline contracts/InterchainTokenService.sol:344]: add test coverage

Comment (ahramy) [reply contracts/InterchainTokenService.sol:344]: added unit tests.

File: test/InterchainTokenService.js [hunk 2]
```diff
@@ -621,6 +621,8 @@ describe('Interchain Token Service', () => {
                     'Call deployInterchainToken on source chain',
                 ),
             )
+                .to.emit(service, 'InterchainTokenIdClaimed')
```
Comment (milapsheth) [inline test/InterchainTokenService.js:624]: also add for the case when deploying on remote chain, and not the current one

File: test/InterchainTokenService.js [hunk 3]
```diff
@@ -846,6 +848,8 @@ describe('Interchain Token Service', () => {
             const params = defaultAbiCoder.encode(['bytes', 'address'], [wallet.address, token.address]);
 
             await expect(reportGas(service.deployTokenManager(salt, '', LOCK_UNLOCK, params, 0), 'Call deployTokenManager on source chain'))
+                .to.emit(service, 'InterchainTokenIdClaimed')
```
Comment (milapsheth) [inline test/InterchainTokenService.js:851]: same

**Review by milapsheth — APPROVED** [review]

File: test/InterchainTokenService.js (hunk 2 above)
Comment (ahramy) [reply test/InterchainTokenService.js:624]: that's good call, updated.

File: test/InterchainTokenService.js (hunk 3 above)
Comment (ahramy) [reply test/InterchainTokenService.js:853]: updated.

**Pull request merged**